### PR TITLE
Added sample code for reactive handling of key rotation.

### PR DESCRIPTION
**Pull request by sapinderpalsingh**
## Purpose
Sample code to demonstrate reactive key rotation handling in .net core applications using polly.
* ...

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ x] Other... Please describe: Sample Code
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,57 +1,65 @@
...
+Sample .net core web api to reactively handle secret key rotation in keyvault.
 
-(short, 1-3 sentenced, description of the project)
+## Pre-Requisites
+You need to have following azure resources
 
-## Features
+* Storage Account
+* Cosmos DB
+* Key Vault
```
Comment (maniSbindra) [inline README.md:14]: Do you think it will be useful adding a section "How it works" somewhere in the document?

File: README.md [hunk 2]
```diff
@@ -1,57 +1,140 @@
-# Project Name
+# Reactive handling of secret key rotation in .Net Core Applications using Polly.
+Secret key rotation is a common process in production applications where organizations do a periodic or sometimes adhoc key rotation of their azure services like cosmos db, storage accounts, event hubs etc. for the security reasons.
 
-(short, 1-3 sentenced, description of the project)
+When the new rotated keys are updated to KeyVault, the consuming apps start failing as the secrets used are not valid anymore, so either we must restart our apps or programmers implement long polling to KeyVault to do a periodic secret refresh. Both approaches are not efficient as the first one requires manual intervention or some sort of custom app that can trigger an app restart when keys get rotated and in the second one if you poll too frequently then it will add to unnecessary KeyVault cost and if you do it with a longer duration then your app can have a downtime.
 
-## Features
+This code sample addresses this problem, i have considered Cosmos and Storage account services as exmples but you can extend it for other services as well.
```
Comment (maniSbindra) [inline README.md]:
```suggestion
This code sample addresses this problem, here Cosmos DB and Storage account services have been considered as examples, but you can extend it for other services as well.
```

Comment (maniSbindra) [reply README.md]: @sapinderpalsingh I would recommend not using "I have" in the sample

**Review by maniSbindra — CHANGES_REQUESTED** [review]
I have requested a couple of minor changes, Thanks

File: README.md (hunk 1 above)
Comment (sapinderpalsingh) [reply README.md:14]: Added this section

File: README.md (hunk 2 above)
Comment (sapinderpalsingh) [reply README.md]: Updated as suggested.

**Review by maniSbindra — APPROVED** [review]
lgtm

**Pull request merged**